src/infill.cpp

```diff
@@ -177,6 +186,35 @@ void Infill::generateCubicSubDivInfill(Polygons& result, const SliceMeshStorage&
     addLineSegmentsInfill(result, uncropped);
 }
 
+void Infill::generateCrossInfill(const SliceMeshStorage& mesh, Polygons& result_polygons, Polygons& result_lines)
+{
+    assert(mesh.cross_fill_pattern);
+    if (zig_zaggify)
+    {
+        outline_offset += -infill_line_width / 2;
+    }
+    coord_t shift = line_distance / 2;
+    bool apply_pockets_alternatingly = false;
+    bool use_odd_in_junctions = false;
+    bool use_odd_out_junctions = false;
+    coord_t pocket_size = 0;
+    if (pattern == EFillMethod::CROSS_3D)
+    {
+        coord_t period = line_distance * 2;
+        shift = z % period;
+        shift = std::min(shift, period - shift); // symmetry due to the fact that we are applying the shift in both directions
+        shift = std::min(shift, period / 2 - infill_line_width / 2); // don't put lines too close to each other
+        shift = std::max(shift, infill_line_width / 2); // don't put lines too close to each other
+
+        apply_pockets_alternatingly = true; // TODO: make this a user setting or decide once and for all which Cross 3D pattern is best.
+        pocket_size = line_distance; // TODO: make user setting
```

Comment:
There's going to be a student project where bachelor students are going to test stiffness of the different structures.

I want to have the cross pattern merged before the next release so that the students can test the infill pattern with a properly tested Cura.

Expect more PRs relating to this infill structure in the 'near' future.

I think I will expose these settings in the experimental category so that the students can experiment with them.

Reply:
Yeah I agree, I think this'll get in before the next release. I'm trying, anyway :wink:

I think for this case it isn't necessary to make them into user settings. But I'm fine with what you changed now too.